Tests/BackendIntegrationTests/StoreKitObserverModeIntegrationTests.swift

```diff
@@ -45,13 +45,13 @@ class StoreKit2ObserverModeIntegrationTests: StoreKit1ObserverModeIntegrationTes
     }
 
     @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
-    func testPurchaseInDevicePostsReceipt() async throws {
+    func testObservingTransactionUnlocksEntitlement() async throws {
         let result = try await self.manager.purchaseProductFromStoreKit2()
         let transaction = try XCTUnwrap(result.verificationResult?.underlyingTransaction)
 
         try self.testSession.disableAutoRenewForTransaction(identifier: UInt(transaction.id))
 
-        XCTExpectFailure("This test currently does not pass (see FB12231111)")
+        _ = try await Purchases.shared.handleObserverModeTransaction(result)
```

Comment:
OMG OMG I missed this! So excited :D 

I think we can simplify the test though. Since this is asynchronous, the check below can be done instantly, right?